src/astro_databricks/operators/notebook.py

```diff
@@ -129,25 +131,58 @@ def _get_task_base_json(self) -> dict[str, Any]:
             "libraries": self.notebook_packages,
         }
 
+    def merge_notebook_packages(self, databricks_task_group: TaskGroup):
+        """
+        Merge the task group notebook packages into the notebook's packages, without adding any identical duplicates.
+        Modifies self.notebook_packages in place.
+
+        Example value for self.notebook_packages:
+        [
+            {"pypi": {"package": "requests_toolbelt==1.0.0"}}
+        ]
+
+        """
+        for task_group_package in databricks_task_group.notebook_packages:
+            exists = False
+            for existing_package in self.notebook_packages:
+                if task_group_package == existing_package:
+                    exists = True
+                    break
+            if not exists:
+                self.notebook_packages.append(task_group_package)
```

Comment:
can we simplify this to something like:
```suggestion
            if not task_group_package in self.notebook_packages:
              self.notebook_packages.append(task_group_package)
```

Reply:
Unfortunately not, because the items of the list are dictionaries, and dictionaries are not hashable in Python.